src/libMessage/Messenger.cpp

```diff
@@ -113,6 +113,20 @@ void NumberToArray(const T& number, vector<unsigned char>& dst,
   Serializable::SetNumber<T>(dst, offset, number, S);
 }
 
+template <class K, class V>
+bool CopyWithSizeCheck(const K& arr, V& result) {
+  LOG_MARKER();
+
+  // Fixed length copying.
+  if (arr.size() != result.max_size()) {
```

Comment:
Why it is not if (arr.size() <= result.max_size()) ?

Reply:
This function will be used for all our fixed-length types (the hashes).  As such, we shouldn't really expect to receive an array that is not exactly the size expected.  So it's better to have the strict `!=` check here.

Reply:
I see